### PR TITLE
Feature/views

### PR DESCRIPTION
**Pull request by ReAliens**
in this pull request, I implemented the following:-
- updated migration to fix foreign keys issues
- add users view page
- add user profile page
- add post list page 
- add post info page 
- style all pages as required in sneak peak

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1 +1,33 @@
-<h1> display certain user posts </h1>
\ No newline at end of file
+<section class="post-section">
+  <% user = User.where(id: params[:user_id])[0] %>
```
Comment (omar-labana) [inline app/views/posts/show.html.erb:2]: - [ ]  I'd suggest moving the logic responsible for getting the user data to its corresponding controller action, controller logic should be separated from rendering views. Should be something like this:
```ruby
@user = User.find(params[:id])
```
then you can access `@user` directly from the view and render its data.

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,39 @@
-<h1> display list of posts </h1>
\ No newline at end of file
+<section class="users-container">
+  <% user = User.where(id: params[:user_id])[0] %>
```
Comment (omar-labana) [inline app/views/posts/index.html.erb]: - [ ] I'd suggest moving the logic responsible for getting the user data to its corresponding controller action, controller logic should be separated from rendering views. Should be something like this:
```ruby
@users = User.all
```
then you can access `@users` directly from the view and render its data.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @ReAliens, 

Excellent work so far! 👌
You have implemented all the required features without any bugs or errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_N/A_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 2 above)
Comment (ReAliens) [reply app/views/posts/index.html.erb]: thanks for your awesome feedback :heart_decoration:

**Review by omar-labana — APPROVED** [review]
# **Approved ✔️**

Hi @ReAliens,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ReAliens** [comment]
thanks for your effort

**Pull request merged**